### PR TITLE
Generate changelog for 3.1.1

### PR DESCRIPTION
**Pull request by sergos**
Also, remove unreleased/ entries.

NO_DOC=changelog
NO_TEST=changelog
NO_CHANGELOG=changelog

### CODE REVIEW

File: changelogs/3.1.1.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+### Replication
+
+* Fixed a bug when a replica could timeout on subscribe if the master had to
+  open a big enough xlog file for that (gh-9094).
+* Fixed a bug that the `box.info.replication[...].downstream.lag` value could be
+  misleading, not updating in time, frozen (gh-9748).
+* Fixed a bug that allowed the old leader in
+  `box.cfg{election_mode = 'candidate'` mode to get re-elected after resigning
+  himself through `box.ctl.demote` (gh-9855).
+* Fixed the inability to add a new replica to the replicaset if the user has
```
Comment (andreyaksenov) [inline changelogs/3.1.1.md:84]:
```suggestion
* Fixed the inability to add a new replica to the replica set if the user has
```

File: changelogs/3.1.1.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+Backported patches from the vanilla LuaJIT trunk (gh-9924). The following issues
+were fixed as part of this activity:
+
+* Fixed `BC_VARG` recording.
+* Fixed `ffi.alignof()` for reference types.
+* Fixed `sizeof()` expression in C parser for reference types.
+* Fixed `ffi.metatype()` for typedefs with attributes.
+* Fixed `ffi.metatype()` for non-raw types.
+* Fixed IR chain invariant in DCE.
+* Fixed OOM errors handling during trace stitching.
```
Comment (andreyaksenov) [inline changelogs/3.1.1.md:104]:
```suggestion
* Fixed OOM error handling during trace stitching.
```

**Review by andreyaksenov — APPROVED** [review]

**Review by Totktonada — APPROVED** [review]
No objections from me.

File: changelogs/3.1.1.md (hunk 1 above)
Comment (sergos) [reply changelogs/3.1.1.md]: done

File: changelogs/3.1.1.md (hunk 2 above)
Comment (sergos) [reply changelogs/3.1.1.md]: done

File: changelogs/3.1.1.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
+# 3.1.1
```
[the remark this answers is not in the document]
Comment (sergos) [reply changelogs/3.1.1.md:1]: done

File: changelogs/3.1.1.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+Date: 2024-07-31
+Tag: 3.1.1
+
+## Overview
+
+3.1.1 is the 2nd [stable][release_policy] version of the 3.1 release
+series. It introduces resolves 36 bugs since 3.1.0.
```
[the remark this answers is not in the document]
Comment (sergos) [reply changelogs/3.1.1.md]: oh... seems reasonable. done.

[the remark this answers is not in the document]
Comment (sergos) [reply changelogs/3.1.1.md]: done.

**Comment by Gumix** [comment]
I guess we should move `box` bullets to the `core` section.

https://github.com/tarantool/tarantool/pull/9953#discussion_r1567128646

**Comment by sergos** [comment]
> I guess we should move `box` bullets to the `core` section.
> 
> [#9953 (comment)](https://github.com/tarantool/tarantool/pull/9953#discussion_r1567128646)

done.

**Pull request merged**